### PR TITLE
COS-93, Ubuntu 18.04 support with Mizar as static pods in scaleout architecture

### PR DESCRIPTION
**Pull request by vinaykul**
**What type of PR is this?**
> Uncomment only one ` /kind <>` line, hit enter to put that in a new line, and remove leading whitespaces from that line:
>
> /kind api-change
> /kind bug
> /kind cleanup
> /kind design
> /kind documentation
> /kind failing-test
/kind feature
> /kind flake

**What this PR does / why we need it**: This is the first step towards getting Mizar deployed and working in Arktos scaleout cluster.

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:
Please review the COS-93 OS changes to see if there's any possibility of regression for current scaleout clusters.
I was able to deploy by calling kube-up.sh with and without NETWORK_PROVIDER=mizar flag.

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

File: cluster/gce/gci/configure-helper.sh [hunk 1]
```diff
@@ -167,6 +169,18 @@ function main() {
...
+#mkdir -p /etc/cni/net.d
+#git clone https://github.com/centaurus-cloud/mizar /var/mizar
+#ln -snf /sys/fs/bpf /bpffs
+#ln -snf /var/mizar/build/bin /trn_bin
+#ln -snf /var/mizar/build/xdp /trn_xdp
+#cp -f /var/mizar/etc/cni/10-mizarcni.conf /etc/cni/net.d/10-mizarcni.conf
+#cp -f /var/mizar/build/bin/mizarcni /opt/cni/bin/mizarcni
+#ln -snf /var/mizar/build/tests/mizarcni.config /etc/mizarcni.config
+#set +x
+
```
Comment (sonyafenge) [inline cluster/gce/gci/configure-helper.sh:183]: This section is commented. if we need this later, Please add comments.

File: cluster/gce/gci/configure.sh [hunk 2]
```diff
@@ -682,6 +750,7 @@ set-broken-motd
 KUBE_HOME="/home/kubernetes"
 KUBE_BIN="${KUBE_HOME}/bin"
 
+ensure-docker
```
Comment (sonyafenge) [inline cluster/gce/gci/configure.sh:753]: this doesn't work for cos OS, at least "apt-get" doesn't work for cos OS

File: cluster/gce/gci/configure-helper.sh (hunk 1 above)
Comment (vinaykul) [reply cluster/gce/gci/configure-helper.sh]: Removed. Using ubuntu for Mizar instead, cos will require a lot more work.

File: cluster/gce/gci/configure.sh (hunk 2 above)
Comment (vinaykul) [reply cluster/gce/gci/configure.sh:761]: fixed to add os-specific installs (supporting ubuntu for now), the rest of the function checks for docker installed, and sets config to overlay2 to work with COS-93.

File: cluster/gce/gci/configure.sh [hunk 3]
```diff
@@ -376,6 +378,10 @@ function install-node-problem-detector {
 function install-cni-network {
   mkdir -p /etc/cni/net.d
   case "${NETWORK_POLICY_PROVIDER:-flannel}" in
+    mizar)
+    setup-mizar-cni-conf
+    install-mizar-cni-bin
+    ;;
```
Comment (sonyafenge) [inline cluster/gce/gci/configure.sh:384]: From: https://github.com/CentaurusInfra/mizar/blob/dev-next/docs/user/getting_started.md, Kernel version need be at least 5.6-rc2 on every node, should we add kernelupdate.sh to Mizar installation?

Comment (vinaykul) [reply cluster/gce/gci/configure.sh:384]: Yes. I prefer that we don't go that route. We should instead try to fix kube-up for 20.04 which has the latest kernel

**Review by sonyafenge — APPROVED** [review]
/LGTM.

Tested regular kube-up and kubemark. no impact for regular kube-up and kubemark.

**Comment by sonyafenge** [comment]
Tested kube-up.sh for scale-up and scale-out, using default GCI_VERSION, NETWORK_PROVIDER. No regression and cluster can be start successfully.

**Comment by vinaykul** [comment]
Re-tested with mizar and ensured that kube-up successfully completes. This brings the first pieces in place, more work is needed. 

Scale-up case:
```bash
export SCALEOUT_CLUSTER=false NUM_NODES=2 RUN_PREFIX=vinay-ark
export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=500GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-16 NODE_DISK_SIZE=500GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=${RUN_PREFIX} KUBE_GCE_NETWORK=${RUN_PREFIX} ENABLE_KCM_LEADER_ELECT=false ENABLE_SCHEDULER_LEADER_ELECT=false ETCD_QUOTA_BACKEND_BYTES=8589934592 SHARE_PARTITIONSERVER=false LOGROTATE_FILES_MAX_COUNT=200 LOGROTATE_MAX_SIZE=200M KUBE_ENABLE_APISERVER_INSECURE_PORT=true KUBE_ENABLE_PROMETHEUS_DEBUG=true KUBE_ENABLE_PPROF_DEBUG=true TEST_CLUSTER_LOG_LEVEL=--v=2 HOLLOW_KUBELET_TEST_LOG_LEVEL=--v=2 GCE_REGION=us-west2-b
NETWORK_PROVIDER=mizar ./cluster/kube-up.sh 
```

Scale-out case:
```bash
export SCALEOUT_CLUSTER=true NUM_NODES=2 SCALEOUT_TP_COUNT=1 SCALEOUT_RP_COUNT=1 RUN_PREFIX=varktos-mizar
export SCALEOUT_CLUSTER=true NUM_NODES=2 SCALEOUT_TP_COUNT=1 SCALEOUT_RP_COUNT=1 RUN_PREFIX=vinay-ark-mizar
export MASTER_DISK_SIZE=500GB MASTER_ROOT_DISK_SIZE=500GB KUBE_GCE_ZONE=us-west2-b MASTER_SIZE=n1-highmem-32 NODE_SIZE=n1-highmem-16 NODE_DISK_SIZE=500GB GOPATH=$HOME/go KUBE_GCE_ENABLE_IP_ALIASES=true KUBE_GCE_PRIVATE_CLUSTER=true CREATE_CUSTOM_NETWORK=true KUBE_GCE_INSTANCE_PREFIX=${RUN_PREFIX} KUBE_GCE_NETWORK=${RUN_PREFIX} ENABLE_KCM_LEADER_ELECT=false ENABLE_SCHEDULER_LEADER_ELECT=false ETCD_QUOTA_BACKEND_BYTES=8589934592 SHARE_PARTITIONSERVER=false LOGROTATE_FILES_MAX_COUNT=200 LOGROTATE_MAX_SIZE=200M KUBE_ENABLE_APISERVER_INSECURE_PORT=true KUBE_ENABLE_PROMETHEUS_DEBUG=true KUBE_ENABLE_PPROF_DEBUG=true TEST_CLUSTER_LOG_LEVEL=--v=2 HOLLOW_KUBELET_TEST_LOG_LEVEL=--v=2 GCE_REGION=us-west2-b
NETWORK_PROVIDER=mizar ./cluster/kube-up.sh
```

**Pull request merged**